src/index.js

```diff
@@ -140,44 +138,42 @@ class ReactCreditCards extends React.Component {
   }
 
   get options() {
-    const { number } = this.props;
-    const issuer = Payment.fns.cardType(number) || 'unknown';
+    const { issuer, number, preview } = this.props;
+    const { validCardTypes } = this.state;
+    let updatedIssuer = issuer || 'unknown';
+
+    if (number && !preview) {
+      const validatedIssuer = creditCardType(sanitizeNumber(number)).length ? creditCardType(sanitizeNumber(number))[0].type : 'unknown';
+
+      if (validCardTypes.includes(validatedIssuer)) {
+        updatedIssuer = validatedIssuer;
+      }
+    }
```

Comment:
I had to tweak the logic here since `creditCardType` will always return an array of card type objects.

Also added the option to bypass this logic if the `preview` prop is passed, in this case, we use the `issuer` prop

Reply:
Another reason for the abstraction.
Get the issuer only if there's a single item in the array.

and we already have an `issuer` getter that handles that.

Reply:
got your point on the `issuer` getter, makes total sense, reverted the logic here.

But I noticed one thing with this part:

> Get the issuer only if there's a single item in the array.

In this case, we get a lot more unknowns, for example:

| number | type |
|--------|------|
|  4  | unknown |
|  40  | unknown |
|  401  | unknown |
|  4011  | unknown |
|  40117  | unknown |
|  401176  | visa |
|  401178  | elo |

Before we got `visa` from `4` until `40117`, and only switched to `elo` if the bin matched. Do you think this could be an issue?